projet/projects/views.py

```diff
@@ -37,4 +38,68 @@ class UsersProjectsList(generics.ListAPIView):
     serializer_class = ProjectSerializer
 
     def get_queryset(self):
-        return Project.objects.filter(user=self.request.user)
\ No newline at end of file
+        return Project.objects.filter(user=self.request.user)
+
+
+class AcceptContributor(generics.RetrieveUpdateDestroyAPIView):
+    serializer_class = ContributorSerializer
+    permission_classes = [IsAuthenticated, IsOwnerOrAdminOfProject]
+    queryset = Contributor.objects.all()
+
+    def perform_update(self, serializer):
+        serializer.save(accepted_by=self.request.user, date_accepted=timezone.now())
+
+    def patch(self, request, project_id=None, contributor_id=None):
+        contributor = Contributor.objects.get(id=contributor_id, project__id=project_id)
+        if request.user == contributor.user:
+            error = "You cannot accept yourself"
+            return Response(error, status=status.HTTP_403_FORBIDDEN)
+
+        contributor.accepted = True
```

Comment:
One should suffice either the perform update or the patch. Will also be changed in a later commit.